File: frontend/src/lib/queryKeys/billing.ts
// Key factory discipline (AD-21) — the 5.6 polling success invalidates the
// plan/packs/history keys on the /billing page.
export const billingKeys = {
  plan: (userKey: string) => ['billing', 'plan', userKey] as const,
  packs: (userKey: string) => ['billing', 'packs', userKey] as const,
  history: (userKey: string) => ['billing', 'history', userKey] as const,
  status: (userKey: string, txnId: string) => ['billing', 'status', userKey, txnId] as const,
}

File: backend/apps/billing/urls/billing.py
from django.urls import path

from apps.billing.views import (
    CancelView,
    CreateCheckoutView,
    HistoryView,
    PacksView,
    PlanView,
    StatusView,
)

app_name = 'billing'

urlpatterns = [
    path('create-checkout/', CreateCheckoutView.as_view(), name='create-checkout'),
    path('plan/', PlanView.as_view(), name='plan'),
    path('packs/', PacksView.as_view(), name='packs'),
    path('history/', HistoryView.as_view(), name='history'),
    path('cancel/', CancelView.as_view(), name='cancel'),
    path('status/<str:txn_id>/', StatusView.as_view(), name='status'),
]
